src/main/webapp/app/exercises/programming/manage/update/switch-edit-mode-button/switch-edit-mode-button.component.ts

```diff
@@ -0,0 +1,20 @@
+import { Component, Input, input } from '@angular/core';
+import { faHandshakeAngle } from '@fortawesome/free-solid-svg-icons';
+import { ArtemisSharedCommonModule } from 'app/shared/shared-common.module';
+import { ButtonSize, ButtonType } from 'app/shared/components/button.component';
+import { ArtemisSharedComponentModule } from 'app/shared/components/shared-component.module';
+
+@Component({
+    selector: 'jhi-switch-edit-mode-button',
+    standalone: true,
+    templateUrl: './switch-edit-mode-button.component.html',
+    imports: [ArtemisSharedCommonModule, ArtemisSharedComponentModule],
+})
+export class SwitchEditModeButtonComponent {
+    protected readonly faHandShakeAngle = faHandshakeAngle;
+
+    @Input() switchEditMode: () => void;
```

Comment:
is the input a function?

Reply:
Yes, as far as I am aware inputs are `readonly`, so I need to pass the method from the parent to allow updating the value